Backend/RoomMe/RoomMe.API/Controllers/HouseworkController.cs

```diff
@@ -0,0 +1,176 @@
+
+            int settingsId = -1;
+
+            if (houseworkEntity == null)
+            {
+                houseworkEntity = housework.ToHouseworkModel();
+                await _sqlContext.Houseworks.AddAsync(houseworkEntity).ConfigureAwait(false);
+                await _sqlContext.SaveChangesAsync().ConfigureAwait(false);
+
+                HouseworkSettings settings = new()
+                {
+                    HouseworkId = houseworkEntity.Id,
+                    FrequencyId = housework.FrequencyId,
+                    Day = housework.Day
+                };
+
+                await _sqlContext.HouseworkSettings.AddAsync(settings).ConfigureAwait(false);
+                await _sqlContext.SaveChangesAsync().ConfigureAwait(false);
+
+                settingsId = settings.Id;
+            }
+            else
+            {
+                houseworkEntity.UpdateHousework(housework, users);
+                _sqlContext.Houseworks.Update(houseworkEntity);
+                await _sqlContext.SaveChangesAsync().ConfigureAwait(false);
+
+                settingsId = houseworkEntity.HouseworkSettings.Id;
+            }
+
+            return houseworkEntity.ToHouseworkPutReturnModel(settingsId);
+        }
+
+        [HttpGet("{houseworkId}/settings", Name = nameof(GetHouseworkSettings))]
+        public async Task<ActionResult<HouseworkSettingsModel>> GetHouseworkSettings(int settingsId)
+        {
+            var settings = await _sqlContext.HouseworkSettings
+                .Include(x => x.Frequency)
+                .FirstOrDefaultAsync(y => y.Id == settingsId)
+                .ConfigureAwait(false);
+
+            if(settings == null)
+            {
+                _logger.LogError($"Settings not found for id {settingsId}");
+                return new BadRequestResult();
+            }
+
+            if(settings.Frequency == null)
+            {
+                _logger.LogError($"Frequency not found for settingsId {settingsId}");
+                return new BadRequestResult();
+            }
+
+            return settings.ToHouseworkSettingsModel();
+        }
+
+        [HttpGet("{houseworkId}/{settingsId}/frequency", Name = nameof(GetSettingsFrequency))]
+        public async Task<ActionResult<HouseworkFrequencyModel>> GetSettingsFrequency(int settingsId)
+        {
+            var settings = await _sqlContext.HouseworkSettings
+                .Include(x => x.Frequency)
+                .FirstOrDefaultAsync(y => y.Id == settingsId)
+                .ConfigureAwait(false);
+
+            if(settings == null)
+            {
+                _logger.LogError($"Housework not found for id {settingsId}");
+                return new BadRequestResult();
+            }
+
+            return settings.Frequency.ToHouseworkFrequencyModel();
+        }
+
+        [HttpGet("list", Name = nameof(GetFullHouseworkByDate))]
+        public async Task<ActionResult<List<HouseworkFullGetModel>>> GetFullHouseworkByDate(FromToDateModel dates)
```

Comment:
To Ci dziala? Powinno rzucac bledem ze GET nie moze miec body

Reply:
Dodatkowo: to jeszcze powinno byc filtrowane po mieszkaniu - chcemy miec tylko prace z jednego mieszkania